hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AbstractAmqpServiceIntegrationTest.java

```diff
@@ -194,11 +196,16 @@ private void assertAssignmentMessage(final Set<SoftwareModule> dsModules, final
     protected void assertDmfDownloadAndUpdateRequest(final DmfDownloadAndUpdateRequest request,
             final Set<SoftwareModule> softwareModules, final String controllerId) {
         Assert.assertThat(softwareModules, SoftwareModuleJsonMatcher.containsExactly(request.getSoftwareModules()));
-        request.getSoftwareModules()
-                .forEach(dmfModule -> assertThat(dmfModule.getMetadata()).containsExactly(
-                        new DmfMetadata(TestdataFactory.VISIBLE_SM_MD_KEY, TestdataFactory.VISIBLE_SM_MD_VALUE)));
+        request.getSoftwareModules().forEach(dmfModule -> assertThat(dmfModule.getMetadata()).containsExactly(
+                new DmfMetadata(TestdataFactory.VISIBLE_SM_MD_KEY, TestdataFactory.VISIBLE_SM_MD_VALUE)));
         final Target updatedTarget = waitUntilIsPresent(() -> targetManagement.getByControllerID(controllerId));
-        assertThat(updatedTarget.getSecurityToken()).isEqualTo(request.getTargetSecurityToken());
+        assert updatedTarget != null;
+        if ((Boolean) anonymousDownloadEnabled.getValue()) {
```

Comment:
Check if it's better/possible to create separate method to  have this condition covered dependent on the goal of the test

Reply:
Behavioural changes in hawkbit were discarded. Test changes not needed anymore